packages/wrangler/src/assets.ts

```diff
@@ -443,6 +443,42 @@ export function validateAssetsArgsAndConfig(
 				"Please remove the asset binding from your configuration file, or provide a Worker script in your configuration file (`main`)."
 		);
 	}
+
+	// Smart placement turned on when using assets
+	if (
+		config?.placement?.mode === "smart" &&
+		config?.assets?.experimental_serve_directly
+	) {
+		logger.warn(
+			"Using assets with smart placement turned on may result in poor performance."
```

Comment:
nit:
```suggestion
			"Using assets with smart placement enabled may result in poor performance."
```

Reply:
any chance we have some docs on this, that we can direct users to so they can better understand what the impact is?

Reply:
I don't think we have any docs on this yet-- @korinne do we want to write something up about smart placement? This will change in Q1 but if we're warning, we might want to drop a link to docs to explain a little more?

Reply:
Based on today's convo, moved out the smart placement pieces to its own PR. Docs included in the draft description:
https://github.com/cloudflare/workers-sdk/pull/7568